lib/galaxy/tools/__init__.py

```diff
@@ -107,6 +108,35 @@ def __init__( self, config_filenames, tool_root_dir, app ):
             tool_root_dir=tool_root_dir,
             app=app,
         )
+        self._view = views.DependencyResolversView(app)
+
+    def tools_requirements(self):
+        reqs = []
+        for desc, tool in self.app.toolbox.tools():
+            for requirement in tool.requirements:
+                if requirement.type == 'package':
+                    reqs.append(json.dumps(requirement.to_dict()))
```

Comment:
You need to sort the dictionary keys here, otherwise 2 identical dicts may be serialized differently:
```python
reqs.append(json.dumps(requirement.to_dict(), sort_keys=True))
```